lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -649,4 +681,151 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
         return String.join(";", refpath);
     }
 
+    private final StatusEvent updateEventForDeletion(final StatusEvent ev)
+              throws RetriableIndexingException,
+                     IndexingException {
+        // wsid and objid of the object that the specified StatusEvent is an event for
+        final int wsid = ev.getAccessGroupId().get();
+        final long objid = Long.valueOf(ev.getAccessGroupObjectId().get()).longValue();
+
+        // check if object is permanently deleted or marked as deleted
+        final Map<String, Object> command;
+        command = new HashMap<>();
+        command.put("command", "listObjects");
+        command.put("params", new ListObjectsParams().
+                withIds(Arrays.asList((long)wsid)).
+                withMinObjectID(objid).
+                withMaxObjectID(objid));
+        try {
+            final List<List> objList = ws.getClient().administer(new UObject(command))
+                    .asClassInstance(List.class);
+
+            if (objList.isEmpty()) {
+                if(ev.getStorageObjectType().isPresent()) {
+                    return StatusEvent.getBuilder(ev.getStorageObjectType().get(),
+                            ev.getTimestamp(),
+                            StatusEventType.DELETE_ALL_VERSIONS).
+                            withNullableAccessGroupID(wsid).
+                            withNullableObjectID(Long.toString(objid)).build();
+                } else {
+                    return StatusEvent.
+                            getBuilder(ev.getStorageCode(),
+                                    ev.getTimestamp(),
+                                    StatusEventType.DELETE_ALL_VERSIONS).
+                            withNullableAccessGroupID(wsid).
+                            withNullableObjectID(Long.toString(objid)).build();
+                }
+            }
+        } catch (IOException ex) {
+            throw handleException(ex);
+        } catch (JsonClientException ex) {
+            if (ex.getMessage().contains("Workspace "+wsid+" is deleted")) {
+                if (ev.getStorageObjectType().isPresent()) {
```

Comment:
Don't think this check is needed as above

Reply:
Might be needed if the workspace was deleted between the time the event was generated and the time the workspace is being indexed?